synapse/handlers/federation.py

```diff
@@ -781,7 +781,7 @@ async def _process_received_pdu(
                     # keys across all devices.
                     current_keys = [
                         key
-                        for device in cached_devices
+                        for device in cached_devices.values()
```

Comment:
I think this might have just been a plain bug, though I don't see it getting hit on matrix.org :/